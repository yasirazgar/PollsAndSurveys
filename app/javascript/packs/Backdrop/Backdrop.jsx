import React from 'react';
import PropTypes from 'prop-types';

import './Backdrop.scss';

const Backdrop = props => (
  <div className="backdrop" onClick={props.backdropClickHandler}/>

);

Backdrop.propTypes = {
  backdropClickHandler: PropTypes.func
}

export default Backdrop;
