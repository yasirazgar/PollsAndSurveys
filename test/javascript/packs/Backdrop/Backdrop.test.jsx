import React from 'react'
import { shallow, mount } from 'enzyme'
import Backdrop from 'packs/Backdrop/Backdrop'

describe("Backdrop", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Backdrop />);

    expect(wrapper).toMatchSnapshot();
  });

  it("sets state sideDrawerOpen to true on clicking sideDrawer toggle-button", () => {
    const clickHandler = jest.fn();
    const wrapper = mount(<Backdrop backdropClickHandler={clickHandler} />)

    wrapper.find('.backdrop').first().simulate('click');

    expect(clickHandler).toHaveBeenCalledTimes(1)
  });
});
