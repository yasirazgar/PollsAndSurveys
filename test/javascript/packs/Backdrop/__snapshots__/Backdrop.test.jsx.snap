// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop renders correctly 1`] = `
<div
  className="backdrop"
/>
`;
